packages/gatsby-theme-store/src/components/Error/ErrorHandler.tsx

```diff
@@ -18,10 +18,14 @@ export const handleError = ({ error, errorId }: Props) => {
   console.error(error)
   console.error(errorId)
 
+  const isUserOffline = !window.navigator.onLine
+
   // prevent infinite loop
   if (
     window.location.pathname.startsWith('/404') ||
-    window.location.pathname.startsWith('/500')
+    window.location.pathname.startsWith('/500') ||
+    window.location.pathname.startsWith('/_v/offline') ||
+    isUserOffline
```

Comment:
why not redirecting the user to the /offline page when the user `isUserOffline` ?